__tests__/app.test.js

```diff
@@ -215,6 +219,64 @@ describe('GET /api/articles', () => {
 			});
 	});
 });
+describe('GET /api/articles - QUERIES', () => {
+	test('200: Accepts a sort-by query and sorts based on input', () => {
+		return request(app)
+			.get('/api/articles/?sort_by=votes')
+			.expect(200)
+			.then((res) => {
+				expect(res.body.articles).toBeSortedBy('votes', {
+					descending: true,
+					coerce: true,
+				});
+			});
+	});
+	test('400: return "Invalid sort by" error when invalid sort by value is passed', () => {
+		return request(app)
+			.get('/api/articles?sort_by=incorrect')
+			.expect(400)
+			.then((res) => {
+				expect(res.body.msg).toBe('Invalid sort by');
+			});
+	});
+	test('200: Accepts a order query and sorts based on input', () => {
+		return request(app)
+			.get('/api/articles/?order=asc')
+			.expect(200)
+			.then((res) => {
+				expect(res.body.articles).toBeSortedBy('created_at', {
+					coerce: true,
+				});
+			});
+	});
+	test('400: return "Invalid order by" error when invalid order value is passed', () => {
+		return request(app)
+			.get('/api/articles?order=incorrect')
+			.expect(400)
+			.then((res) => {
+				expect(res.body.msg).toBe('Invalid order by');
+			});
+	});
+	test('200: Accepts a topic query and filters based on input', () => {
+		return request(app)
+			.get('/api/articles/?topic=cats')
+			.expect(200)
+			.then((res) => {
+				expect(res.body.articles.length).toBe(1);
+				res.body.articles.forEach((article) => {
+					expect(article.topic).toBe('cats');
+				});
+			});
+	});
+	test('400: return "Invalid query parameter" error when invalid topic value is passed', () => {
+		return request(app)
+			.get('/api/articles?topic=incorrect')
+			.expect(400)
```

Comment:
this one should be a 404, topic not found. As the data type is still correct, the topic just doesn't exist